src/store/modules/web3.ts

```diff
@@ -61,7 +61,7 @@ const actions = {
   },
   loadProvider: async ({ commit, dispatch }) => {
     try {
-      if (auth.provider.removeAllListeners) auth.provider.removeAllListeners();
+      if (auth.web3.removeAllListeners) auth.web3.removeAllListeners();
```

Comment:
The listeners are in `auth.provider` if you use `removeAllListeners()` on `auth.web3` the listeners don't get removed. You can try to connect with Metamask with your Account 1 then switch to account 2 then 3, you will see `accountsChanged` hook is called multiple time while there should be only one call.

Reply:
Yes, I noticed that yesterday and was wondering why it was happening. So then I'd suggest we skip that line when `isTorus` is `true`

Reply:
@samuveth Yes you can do that